### ISSUE TITLE
Allow user to specify latest, preview, and none in AnalysisLevel

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn-analyzers#3996]
## latest

- `EnableNETAnalyzers` is set to `true`
- `Microsoft.CodeAnalysis.NetAnalyzers.targets` sets `_GlobalAnalyzerConfigFile` to point to current analyzers `.editorconfig`
- `WarningLevel` is set to `5`

## preview

- `EnableNETAnalyzers` is set to `true`
- `Microsoft.CodeAnalysis.NetAnalyzers.targets` sets `_GlobalAnalyzerConfigFile` to point to latest analyzers `.editorconfig`
- `WarningLevel` is set to `9999`

## none

- `EnableNETAnalyzers` is set to `false`
- `WarningLevel` is set to `4`

**Comment by mikadumont** [comment]
Do we still need a preview option?

**Comment by jmarolf** [comment]
The compiler has a `preview` option for opting into new features without installing a preview. The advantage is that we can get users to easily try new things we add without installing a whole new sdk.

**Comment by mavasani** [comment]
Looks good to me!

**Comment by jmarolf** [comment]
I would add. The compiler team plans to add new warning waves in-between releases of .NET. If users want to opt into these new warnings they would need some concept of `preview`.

**Comment by jmarolf** [comment]
@mavasani so I am thinking that we would generate a `.editorconfig` file per analysis level. how difficult would that be to do with the infra for this repo?

**Comment by mavasani** [comment]
@jmarolf We already generate a global editorconfig file for each AnalyzerRelease specified in https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md. For example, if you look at the latest NuGet package, we have the following in the `build\config` folder:

![image](https://user-images.githubusercontent.com/10605811/89948564-32832e00-dbdb-11ea-93a6-e827cf080bf4.png)

![image](https://user-images.githubusercontent.com/10605811/89948644-534b8380-dbdb-11ea-9c68-4c4fc2d84acc.png)

We also drop in a target that can map from an MSBuild property `MicrosoftCodeAnalysisNetAnalyzersRulesVersion` to specific global editorconfig and add it to the command line:
![image](https://user-images.githubusercontent.com/10605811/89948797-94dc2e80-dbdb-11ea-9b45-1f1c744a1162.png)

So, the only missing piece is mapping `AnalysisLevel` to `MicrosoftCodeAnalysisNetAnalyzersRulesVersion`. I plan to do the following:

1. Update https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md to move all the current CA rules (those currently in that shipped file + unshipped file) to the shipped file with a release version `5`, instead of `2.9.8` that we have right now in that file. Basically each supported `AnalysisLevel` value will map to an equivalent `Release` entry in `AnalyzerReleases.Shipped.md`.
2. Default `MicrosoftCodeAnalysisNetAnalyzersRulesVersion` value to `AnalysisLevel`. This will ensure `AnalysisLevel` will automatically map to the correct global editorconfig file and pass it to the compiler.
3. Setting `AnalysisLevel` to `latest` would mean no global editorconfig is chosen, you get everything that is in the package with their defaults.
4. We probably need to figure out a special way to treat `preview` as a special all but last release before `latest`. I will leave that to a follow-up PR.

In short, this should be a simple change that I plan to do soon. `AnalyzerReleases.Shipped.md` will serve dual purpose of warning wave support + documenting released analyzers with their default severities/enabled by default values with each release. We have strict build enforcement that each new CA analyzer added to the repo must have an entry in `AnalyzerReleases.Shipped.md` or `AnalyzerReleases.Unshipped.md`, so things cannot go out-of-date here when new analyzers are added.

**Comment by mavasani** [comment]
@jmarolf @mikadumont https://github.com/dotnet/roslyn-analyzers/pull/3997 adds the end-to-end support for analyzer warning waves based on `AnalysisLevel` property.
